lib/nats.js

```diff
@@ -401,7 +444,7 @@ Client.prototype.setupHandlers = function() {
     // Only bubble up error if we never had connected
     // to the server and we only have one.
     if (client.wasConnected === false && client.servers.length === 0) {
-      client.emit('error', 'Could not connect to server: ' + exception);
+      client.emit('error', new NatsError(CONN_ERR_MSG_PREFIX + exception, CONN_ERR, exception));
```

Comment:
The issue is the that event handler will be called, and now they have to test if the error has the property or not. With the current form, the errors are uniform, but in the case of the connection they can look to see if there's additional information.